src/pani/async.cljs

```diff
@@ -0,0 +1,83 @@
+(ns pani.async
+  (:require [pani.core :as p]
+            [cljs.core.async :refer [<! >! chan put! close!]])
+  (:require-macros [cljs.core.async.macros :refer [go go-loop]]))
+
+(defn with-chan
+  "Call a function with a fresh channel, then return the channel" [f]
+  (let [ch (chan)] (f ch) ch))
+
+(defn chan->cb
+  "Create callback that pushes arguments onto given chan, and closes on nil"
+  [ch]
+  (fn [val] (if val (put! ch val))))
```

Comment:
Happy where it is, so similar to its sibling function which preambles the main family of functions :family: